fluentd/run.sh

```diff
@@ -101,6 +104,33 @@ else
     rm -f $CFG_DIR/openshift/filter-pre-mux-client.conf
 fi
 
+if [ "${USE_MUX:-}" = "true" ] ; then
+    TOTAL_MEMORY_LIMIT=`echo $MUX_MEMORY_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+else
+    TOTAL_MEMORY_LIMIT=`echo $FLUENTD_MEMORY_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+fi
+BUFFER_SIZE_LIMIT=`echo $BUFFER_SIZE_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc`
+if [ $BUFFER_SIZE_LIMIT -eq 0 ]; then
+    BUFFER_SIZE_LIMIT=1048576
+fi
+
+DIV=1
+if [ "$ES_HOST" != "$OPS_HOST" ] || [ "$ES_PORT" != "$OPS_PORT" ] ; then
+    # using ops cluster
+    DIV=`expr $DIV \* 2`
+fi
+if [ "${USE_MUX_CLIENT:-}" = "true" ] ; then
+    DIV=`expr $DIV \* 2`
+fi
+
+# MEMORY_LIMIT per buffer
+MEMORY_LIMIT=`expr $TOTAL_MEMORY_LIMIT / $DIV`
+BUFFER_QUEUE_LIMIT=`expr $MEMORY_LIMIT / $BUFFER_SIZE_LIMIT`
+if [ $BUFFER_QUEUE_LIMIT -eq 0 ]; then
```

Comment:
What are the scenarios where `BUFFER_QUEUE_LIMIT` becomes zero?  A quick test shows that `expr 1 / 2` returns `0` in bash.  If this then is the case where the `BUFFER_SIZE_LIMIT` (which is the fluentd `buffer_chunk_limit`, I believe) is greater than it would seem we have a configuration error, and we should not default to `1024` for the queue size.  That might be outside of the supported memory configuration.